### PR TITLE
osd/OSD: num_running > 0 ensure that slot can't remove.

### PR DESCRIPTION
**Pull request by majianpeng**
Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by majianpeng** [comment]
retest this please

**Comment by majianpeng** [comment]
@neha-ojha please review this.

**Review by neha-ojha — COMMENTED** [review]
could you please add the rationale behind this change in your commit description?

**Comment by majianpeng** [comment]
@neha-ojha . update by your suggestion. please review.

**Comment by majianpeng** [comment]
retest this please

**Comment by majianpeng** [comment]
@neha-ojha  ping

**Comment by neha-ojha** [comment]
> @neha-ojha ping

@majianpeng apologies about the delay and thanks for adding the explanation, I will review it this week. thanks!

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -10543,14 +10543,8 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb)
     sdata->shard_lock.lock();
 
     auto q = sdata->pg_slots.find(token);
-    if (q == sdata->pg_slots.end()) {
-      // this can happen if we race with pg removal.
```
Comment (neha-ojha) [inline src/osd/OSD.cc:10547]: why does this comment not hold true any longer? why can we not race with pg removal?

**Pull request closed without merging**